sample_size/sample_size_calculator.py

```diff
@@ -80,3 +81,13 @@ def get_sample_size(self) -> float:
                 sample_size = self._get_single_sample_size(ratio_metric)
 
         return sample_size
+
+    def register_metric(self, metric_type: str, metric_metadata: Dict[str, float]) -> None:
+        VAR_REGISTER_FUNC_MAP = {
+            "boolean": "register_bool_metric",
+            "numeric": "register_numeric_metric",
+            "ratio": "register_ratio_metric",
```

Comment:
when this function is used by the script only, hardcoded the string key here seems fine. but if now we move it here, i'm wondering if there is a way to let the input metric_type always synced. e.g if anytime we rename "boolean" to "bool" in line 87, all apps will break.

one way i have in mind is to generate a constant in this file
```python
class MetricType:
    BOOL = "boolean"
    NUMERIC = "numeric"
    RATIO = "ratio"
```
Then utils.py could create another constant like below used in [line 65](https://github.com/godaddy/sample-size/blob/063d2748808321080aa8b88c44c926c6f95c4ed7/sample_size/scripts/utils.py#L65) by import MetricType 
```
METRIC_TYPES = [MetricType.BOOL, MetricType.RATIO, MetricType.NUMERIC]
```

and any other app could also do something like `calculator.register_metric(MetricType.BOOL, ..)`

But I don't know if that really matters at all considering if we actually rename, we will publish a new version anyway and it is probably unlikely we will rename though